src/gun_http.erl

```diff
@@ -831,12 +831,12 @@ down(#http_state{streams=Streams}) ->
 error_stream_closed(State, StreamRef, ReplyTo) ->
 	ReplyTo ! {gun_error, self(), stream_ref(State, StreamRef), {badstate,
 		"The stream has already been closed."}},
-	State.
+	{state, State}.
 
 error_stream_not_found(State, StreamRef, ReplyTo) ->
 	ReplyTo ! {gun_error, self(), stream_ref(State, StreamRef), {badstate,
 		"The stream cannot be found."}},
-	State.
+	{state, State}.
```

Comment:
I will change these two functions to return `ok` and move the calls out in order to return `[]` instead. Will be clearer.

Reply:
Sure please do I'm eating breakfast.

Reply:
Done. Btw, have you considered using GitHub's "Squash merge" button? IMO it's practical to keep multiple commits within a PR just in case you need to split partial changes into a separate PR. ;-)

Reply:
Thanks! I'm on it.

I don't merge via GitHub's UI. The upstream repository is on my laptop, not on GitHub. I rebase and amend when necessary when I merge. I will probably keep the stream_* functions change in a separate commit anyway because it is a different change.